R/further_analytical_functions.R

```diff
@@ -0,0 +1,308 @@
+#'
+#' \itemize{
+#'   \item \bold{TFR}:  Total fertility rate.
+#'   \item \bold{GRR}:  Gross reproduction rate.
+#'
+#'                    \eqn{GRR = TFR * 0.488}
+#'
+#'   \item \bold{NRR}: Net reproduction rate.
+#'
+#'                    \eqn{NRR = sum(GRR * age specific fertility * age specific survival / 10000)}
+#'
+#'   \item \bold{r}:  Intrinsic growth rate in percent per year.
+#'
+#'                    \eqn{r = 100 * log(NRR) / generation length}
+#'
+#'   \item \bold{Dt}: Doubling time in years.
+#'
+#'                    \eqn{Dt = 100 * 0.6931 / r}
+#'  }
+#'
+#' @examples
+#' # Calculate reproduction indices from real life dataset.
+#' library(magrittr)
+#' mag <- magdalenenberg
+#' mag <- mag %>% replace(mag == "60-x", "60-69")
+#' mag <- mag %>% tidyr::separate(a, c("from", "to")) %>%
+#' transform(from = as.numeric(from), to = as.numeric(to))
+#' mag_prep <- mag %>% prep.life.table(dec = "Dx",
+#' agebeg = "from", ageend = "to", method = "Equal5",
+#' agerange = "included")
+#' mag_result <- mag_prep %>% life.table()
+#' lt.reproduction(mag_result)
+#
+#' @export
+#'
+lt.reproduction <- function(life_table, fertility_rate = "BA_log",  gen_len = 20) {
+
+  indx <- lt.indices(life_table)
+
+  if (fertility_rate == "McO") {
+    fertil_rate <- indx$D0_14_D[[1]] * 7.734 + 2.224
+  } else if (fertility_rate == "BA_linear"){
+    # Linear regression
+    fertil_rate <- indx$p5_19[[1]] * 25.7557 + 2.85273
+  } else if (fertility_rate == "BA_power"){
+    # power fit
+    fertil_rate <- 13.4135 * (indx$p5_19[[1]])**0.374708
+  } else if (fertility_rate == "BA_log"){
+    # logarithmic fit
+    fertil_rate <- log(indx$p5_19[[1]]) * 1.58341 + 9.45601
+  } else if (length(fertility_rate) > 0 & is.numeric(fertility_rate)) {
+    fertil_rate <- fertility_rate
+  } else {
+    paste0(
+      "Please choose a valid fertility rate (either 'McO', 'BA_linear', 'BA_log' or 'BA_power' or a number)."
+    ) %>%
+      message
+  }
+
+  # Gross reproductive rate only for females according to Hassan
+  R_pot_fem <- fertil_rate * 0.488
+
+  # Net reproduction rate after Hassan
+  fertility_perc <- c(10.9, 23.1, 24.1, 19.5, 14.4, 6.2, 1.6)
+  R_0 <- (c(indx$repro_indx$fertil_lx) * c(fertility_perc * R_pot_fem / 10000)) %>% sum
+
+  # Intrinsic growth rate in percent per year after Hassan
+  intr_grow <- 100 * log(R_0) / gen_len
+
+  # Doubling time in years after Hassan
+  Dt <- 100 * 0.6931 / intr_grow
+
+  fertiliy_rate <- c(round(fertil_rate, 1), "Total fertility rate")
+  R_pot_fem <- c(round(R_pot_fem, 1), "Gross reproduction rate")
+  R_0 <- c(round(R_0, 2), description = "Net reproduction rate")
+  intr_grow <- c(round(intr_grow, 2), "Intrinsic growth rate (perc/y)")
+  Dt <- c(round(Dt, 1), "Doubling time in years")
+  result <- data.frame(rbind(fertiliy_rate, R_pot_fem, R_0, intr_grow, Dt))
+  colnames(result) <- c("value", "description")
+  rownames(result) <- c("TFR","GRR", "NRR","r", "Dt")
+  format(result, justify = "left")
```

Comment:
Not necessary. `return(result)` is much better readable.

Do not care too much about how something might look like on somebodies console. You can't control that anyway. But that's just my opinion.

Reply:
Done.